contrib/mod_sftp/kex.c

```diff
@@ -4148,7 +4154,7 @@ static const unsigned char *calculate_curve25519_h(struct sftp_kex *kex,
   unsigned char *buf, *ptr;
   uint32_t buflen, bufsz;
 
-  bufsz = buflen = 4096;
+  bufsz = buflen = SFTP_KEX_H_BUFLEN;
```

Comment:
In the case of this particular ticket/client/configuration, this 4K buffer size was not large enough for the combined KEXINIT messages from Cyberduck and ProFTPD, both supporting many different algorithms.